dbtmetabase/metabase.py

```diff
@@ -404,14 +434,192 @@ def build_metadata_lookups(
+                    header = "### Dashboard Cards: {}\n\n".format(
+                        str(len(dashboard["ordered_cards"]))
+                    )
+                    for dashboard_model in dashboard["ordered_cards"]:
+                        dashboard_model_ref = dashboard_model.get("card", {})
+                        if not "id" in dashboard_model_ref:
+                            continue
+                        self._extract_card_exposures(dashboard_model_ref["id"])
+                else:
+                    continue
+
+                # No spaces allowed in model names in dbt docs DAG / No duplicate model names
+                name = name.replace(" ", "_")
+                enumer = 1
+                while name in duplicate_name_check:
+                    name = f"{name}_{enumer}"
+                    enumer += 1
+                duplicate_name_check.append(name)
+
+                # Construct Exposure
+                description = (
+                    header
+                    + (
+                        "{}\n\n".format(description.strip())
+                        if description
+                        else "No description provided in Metabase\n\n"
+                    )
+                    + (
+                        "#### Query\n\n```\n{}\n```\n\n".format(
+                            "\n".join(
+                                line
+                                for line in self.native_query.strip().split("\n")
+                                if line.strip() != ""
+                            )
+                        )
+                        if self.native_query
+                        else ""
+                    )
+                    + "#### Metadata\n\n"
+                    + ("Metabase Id: __{}__\n\n".format(item["id"]))
+                    + ("Created On: __{}__".format(created_at))
+                )
+                captured_exposure = {
+                    "name": name,
+                    "description": description,
+                    "type": "analysis" if item["model"] == "card" else "dashboard",
+                    "url": f"{self.protocol}://{self.host}/{item['model']}/{item['id']}",
+                    "maturity": "medium",
+                    "owner": {
+                        "name": creator_name,
+                        "email": creator_email,
+                    },
+                    "depends_on": [
+                        refable_models[exposure.upper()]
+                        for exposure in list({m for m in self.models_exposed})
+                        if exposure.upper() in refable_models
+                    ],
+                }
+                captured_exposures.append(captured_exposure)
+
+        # Output dbt YAML
+        with open(
+            os.path.expanduser(os.path.join(output_path, f"{output_name}.yml")), "w"
+        ) as docs:
+            yaml.dump(
+                {"version": 2, "exposures": captured_exposures},
```

Comment:
Should the version be a constant somewhere? E.g. if dbt bumps it up in future.

Reply:
adjusted to be constant at top of func